e2e/wsl-integrations.e2e.spec.ts

```diff
@@ -210,83 +210,94 @@ test.describe('WSL Integrations', () => {
     expect(wslIntegrationList.getByText('alpha')).not.toBeNull();
     expect(wslIntegrationList.getByText('beta')).not.toBeNull();
     expect(wslIntegrationList.getByText('gamma')).not.toBeNull();
-  });
 
-  /*
-  test('should show checkbox states', (async() => {
-    const integrations = wslPage.wslIntegrations;
-    const alpha = integrations.find(item => item.name === 'alpha');
-    const beta = wslPage.getIntegration('beta');
-    const gamma = wslPage.getIntegration('gamma');
-
-    await expect(alpha.locator).toHaveCount(1);
-    await expect(alpha.checkbox).not.toBeChecked();
-    await expect(alpha.name).toHaveText('alpha');
-    await expect(alpha.error).not.toBeVisible();
-
-    await expect(beta.locator).toHaveCount(1);
-    await expect(beta.checkbox).toBeChecked();
-    await expect(beta.name).toHaveText('beta');
-    await expect(beta.error).not.toBeVisible();
-
-    await expect(gamma.locator).toHaveCount(1);
-    await expect(gamma.checkbox).not.toBeChecked();
-    await expect(gamma.name).toHaveText('gamma');
-    await expect(gamma.error).toHaveText('some error');
+    expect(await wslPage.alpha.isChecked()).toBeFalsy();
+    expect(await wslPage.beta.isChecked()).toBeTruthy();
+    expect(await wslPage.gamma.isChecked()).toBeFalsy();
+
+    const craftyErrorMessage = 'Error: some error';
+    let parent = wslPage.page.locator('[data-test="item-alpha-parent"]');
+
+    await expect(parent.filter({ hasText: craftyErrorMessage })).toHaveCount(0);
+    parent = wslPage.page.locator('[data-test="item-beta-parent"]');
+    await expect(parent.filter({ hasText: craftyErrorMessage })).toHaveCount(0);
+    parent = wslPage.page.locator('[data-test="item-gamma-parent"]');
+    await expect(parent.filter({ hasText: craftyErrorMessage })).toHaveCount(1);
   });
 
   test('should allow enabling integration', async() => {
+    // This is how we do a reload...
```

Comment:
I don't think we're doing a reload here, just casting `preferencesWindow` (which is a `Page`) to a `Page` subclass that knows how to handle the specifics of that window, and then grabbing the `wslPage` off it.